drivers/modem/quectel-bg9x.c

```diff
@@ -17,6 +20,9 @@ static struct modem_data       mdata;
 static struct modem_context    mctx;
 static const struct socket_op_vtable offload_socket_fd_op_vtable;
 
+#define IO_EXPANDER_NODE        DT_NODELABEL(pi4ioe5v9)
```

Comment:
The IO Extender is not part of the Quectel chip, so its code does not belong in here. It should be a separate entity.
Note that the quectel is connected to an io extender is a feature of the board it is designed into, not a feature of the quectel driver itself.